### DISCUSSION TITLE
Curriculum with go as a backend

### DISCUSSION THREAD

**Discussion by danush754** [TheOdinProject/curriculum#29088] [Ideas]
Hello everyone, why can't we start a curriculum with golang like we have in ruby ?

**Comment by mao-sz** [comment]
TOP's focus has always been programming fundamentals in the context of web dev and less about learning specific technologies themselves. The tech stacks in the curriculum are just vehicles for teaching those programming fundamentals.

The curriculum already provides two journeys to reach the same destination but even if the backend stacks are different, the focus is still about learning core fundamental skills that can be transferred as needed. Having this option is nice but there are hugely diminishing returns the more options you add:

- Workload in producing a whole pathway is *huge*, like really huge. It's not a small thing by any means, especially if you want to (and should want to) have it up to the same standard as the existing pathways.
- You can't just make it and leave it. You need to maintain it, meaning you need a group of experienced people to maintain that pathway. The team are already very bogged down with maintaining and improving the current curriculum.
- The more options you provide, the more splintered the community becomes and the less people are able to help each other. The community is therefore less effective at supporting people through the curriculum.
- The more options you provide, the more you're focusing on teaching the individual tech rather than the core programming skills which you can do in any language, and anyone learning well from them can use to learn whatever they need in their own time. It'd be very surprising if someone completed TOP yet was unable to make use of documentation and own practice to learn new things as needed. Fundamentals are transferable, like you don't necessarily need to learn how to drive in this specific car. You can learn in any car then your driving skills help you learn how to drive a different car far more easily. Many people who get dev jobs after TOP don't work with the exact same tech stack/language as what they learnt with.

If you really really really want to learn Golang and only Goland, not anything else, TOP won't be for you then. Otherwise, I'd be very surprised if someone completed TOP and wasn't able to use their fundamental skills to find their own way through learning how to do things with Golang.
